In [37]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [38]:
df_original = pd.read_csv('../data/ej3_tabla_p2.csv')
df = df_original.copy()

df

EDAD RIESGO EXAMEN
0    55   ALTO     SI
1    56  MEDIO     NO
2    58  MEDIO     SI
3    56   BAJO     NO
4    59   BAJO     NO
5    57  MEDIO     SI
6    60   BAJO     SI
7    53   ALTO     NO
8    59  MEDIO     SI

In [39]:
df['RIESGO'] = df['RIESGO'].map({'BAJO': 1, 'MEDIO': 2, 'ALTO': 3})
df['EXAMEN'] = df['EXAMEN'].map({'SI': 1, 'NO': 0})

In [40]:
print('Correlacion entre edad y riesgo:', df['EDAD'].corr(df['RIESGO']))


Correlacion entre edad y riesgo: -0.7150969419341943


Interpretación (r = −0.715):

- Es una correlación lineal negativa moderada–fuerte.

- Con tu codificación RIESGO: BAJO=1, MEDIO=2, ALTO=3, el signo “−” indica que a mayor EDAD, menor RIESGO (los más jóvenes tienden a tener riesgo más alto).

- La magnitud (~0.72) sugiere una relación importante: si ajustaras una recta, r² ≈ 0.51 ⇒ ~51% de la variación (numérica) del riesgo quedaría explicada por la edad bajo este modelo lineal.

In [41]:
from sklearn.preprocessing import MinMaxScaler
edad_scaler = MinMaxScaler().fit(df[['EDAD']])
df['EDAD'] = edad_scaler.transform(df[['EDAD']])


In [42]:
test_df = q = pd.DataFrame({"EDAD":[50, 51, 54],
                  "RIESGO":["BAJO","MEDIO","ALTO"]})

test_df['RIESGO'] = test_df['RIESGO'].map({'BAJO': 1, 'MEDIO': 2, 'ALTO': 3})
test_df['EDAD'] = edad_scaler.transform(test_df[['EDAD']])

X_test = test_df[['EDAD', 'RIESGO']].to_numpy()


print("X_test shape:", X_test.shape)

print("X_test:", X_test)







X_test shape: (3, 2)
X_test: [[-0.42857143  1.        ]
 [-0.28571429  2.        ]
 [ 0.14285714  3.        ]]


In [43]:
weights = np.array([0.0807, 0.074])
bias = -0.0742


y_pred = X_test @ weights + bias
score = (y_pred > 0).astype(int)

print(y_pred)
print(score)









[-0.03478571  0.05074286  0.15932857]
[0 1 1]
